src/features/ci-cd/hooks/service-installs.ts

```diff
@@ -260,6 +260,13 @@ hooks.addPureHook('POST', 'resin', 'device', {
 		const app = request.values.belongs_to__application;
 		if (app != null) {
 			await createAppServiceInstalls(rootApi, app, [deviceId], tx);
+			if (ASYNC_TASKS_ENABLED && ASYNC_TASK_CREATE_SERVICE_INSTALLS_ENABLED) {
```

Comment:
Shouldn't `ASYNC_TASK_CREATE_SERVICE_INSTALLS_ENABLED` only ever equal `true` if `ASYNC_TASKS_ENABLED` is also `true`? Maybe there was a discussion about this at some point, but I feel like system init should fail if `ASYNC_TASKS_ENABLED === false && ASYNC_TASK_CREATE_SERVICE_INSTALLS_ENABLED === true`.

Reply:
Yea I agree on that this was strange. I expected it to only have to check one of the two, maybe something like:
```
export let ASYNC_TASK_CREATE_SERVICE_INSTALLS_ENABLED = ASYNC_TASKS_ENABLED && boolVar(
	'ASYNC_TASK_CREATE_SERVICE_INSTALLS_ENABLED',
	false,
);
```
But since I don't have background on why it was done this way, I just copied it "as is" from above where we do the same check. I guess that it might be this way to make mocking in tests easier. I would prefer to PR such a change separately and fix all occurrences and whichever tests might break after having a short chat on it.

Reply:
This makes sense and it is an improvement on its own, thanks for the suggestion.